apps/api/src/domain/models/media-item/node-media-management.service.ts

```diff
@@ -0,0 +1,259 @@
+        @Inject(MEDIA_PROBER_TOKEN) private readonly mediaProber: IMediaProber,
+        @Inject(ID_MANAGER_TOKEN) private readonly idManager: IIdManager,
+        private readonly commandHandlerService: CommandHandlerService
+    ) {}
+
+    create(_binary: ReadableStream): Promise<ResultOrError<MediaCreationAcknowledgement>> {
+        throw new Error('Creating a media item from binary is not yet supported.');
+    }
+
+    /**
+     * Note that this requires an admin to first transfer the files to an ingestion
+     * directory.
+     *
+     * @param filepath path to the file within an ingestion directory accessible to the node process
+     */
+    async discover(filepath: string): Promise<ResultOrError<MediaCreationAcknowledgement>> {
+        if (!existsSync(filepath)) {
+            return new FileNotFound(filepath);
+        }
+
+        const extension = path.extname(filepath);
+
+        if (!isNonEmptyString(extension)) {
+            throw new InternalError(
+                `encountered an invalid filepath: ${filepath}. must include an extension`
+            );
+        }
+
+        const dirname = path.dirname(filepath);
+
+        const filename = path.basename(filepath);
+
+        if (!isNonEmptyString(extension)) {
+            throw new InternalError(
+                `failed to parse extension for file: ${filepath}. Invalid result: ${extension}`
+            );
+        }
+
+        // does this throw if it doesn't recognize the extension?
+        const expectedMimetypeFromExtension = getExpectedMimeTypeFromExtension(extension);
+
+        const probeResult = await this.mediaProber.probe(filepath);
+
+        if (isNotFound(probeResult)) {
+            throw new Error('todo- return this error');
+        }
+
+        /**
+         * TODO it is critical that we validate the mime type against the extension
+         * before opening up an endpoint to upload binary files
+         */
+        // if (expectedMimetypeFromExtension !== probeResult.mimeType) {
+        //     throw new Error(`todo = return this error?`);
+        // }
+
+        // TODO Execute a command instead
+        const newMediaItem: MediaItem = new MediaItem({
+            // TODO shouldn't this be passed in?
+            title: dirname,
+            // TODO remove this
+            type: ResourceType.mediaItem,
+            mimeType: expectedMimetypeFromExtension,
+            // this is totally wrong!
+            published: true,
+            // TODO we need the ID generator- better to just execute the command
+            id: buildDummyUuid(1),
+            // todo remove this!
+        });
+
+        /**
+         * TODO We may want to pass this state in from above.
+         * Consider generating many for discoverAll
+         */
+        const generatedId = await this.idManager.generate();
+
+        /**
+         * There's a bit of a chicken-and-egg problem here. The command handler
```

Comment:
Is this still a problem?

Reply:
No. Encapsulating the use of the command handler service solve the problem. The media service manages discovering the metadata first and then building and executing the command.